### PR TITLE
fix(lights): Fixed Lighting Auto Config on spawn

### PR DESCRIPTION
**Pull request by Maximilian-Reuter**
Fixes #[issue_no]

## Summary of Changes

This PR fixes the auto lighting config, that tries to set the brightness of the cockpit screens and the dome, flood and integral lights based on the ambient brightness in the sim.

This now leads to dome lights, flood light and integral lights to be on when spawning at night or dusk and dawn.

For people who don't like the auto lighting config, they can use the EFB light presets to load their own preset automatically.

## Additional context

Discord username (if different from GitHub):\_chaoz_

## Testing instructions

1. Spawn either C/D
2a. IF not DAY: Capt dome light should be on dim
2b. IF DAY: nothing should be on
3. Turn on BAT and GPU
4a. IF not DAY: floods should be at 20% and integral lights should be on and screens at either 15% or 35% depending on how dark it is
4b. IF DAY: Only screens should come on at 85% brightness

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

**Comment by tracernz** [comment]
We have code already that sets the brightness all based on the ambient light level but it seems to not be working. Better to fix that instead.

**Comment by Maximilian-Reuter** [comment]
> We have code already that sets the brightness all based on the ambient light level but it seems to not be working. Better to fix that instead. It probably needs to wait for the GameStateProvider to say we're ingame, as the .flt files are probably overriding it if it's too quick.

but what would the knobs then do ?

**Comment by tracernz** [comment]
> 
> but what would the knobs then do ?
> 
> EDIT: also doesnt that code only work on screens ? Not the integ lights or the dome light ?

It's supposed to set everything once on spawn, so the knobs are still good. It does do integ lights, not but not dome.

**Comment by becas22** [comment]
> I have had multiple people complain about having to always turn up the brightness on every screen and also remark that it feels unrealistic as screens don't need to be turned down anymore since they aren't CRTs.

FYI, the parking flow still orders to dim the DUs

File: fbw-a32nx/src/systems/systems-host/index.ts [hunk 1]
```diff
@@ -30,6 +32,12 @@ class SystemsHost extends BaseInstrument {
         this.hEventPublisher = new HEventPublisher(this.bus);
         this.powerSupply = new PowerSupplyBusses(this.bus);
         this.atsu = new AtsuSystem(this.bus);
+        KeyEventManager.getManager(this.bus).then((manager) => {
+            this.keyInterceptManager = manager;
+            if (this.gameState === 3) {
```
Comment (tracernz) [inline fbw-a32nx/src/systems/systems-host/index.ts:37]:
```suggestion
            if (this.gameState === GameState.ingame) {
```

**Review by tracernz — COMMENTED** [review]
You should probably really await both GameStateProvider and KeyEventManager before calling initLighting. There should then be no need to call it in the update loop.

**Review by tracernz — APPROVED** [review]

**Comment by Maximilian-Reuter** [comment]
Got multiple reports that it works so I will merge

**Pull request merged**